code-challenges.test.js

```diff
@@ -15,42 +15,139 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("gibberishMaker", () => {
+    it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+        const secretCodeWord1 = "Lackadaisical"
+            // Expected output: [-8, 90]
+        const secretCodeWord2 = "Gobbledygook"
+            // Expected output: [5, 109]
+        const secretCodeWord3 = "Eccentric"
+      expect(gibberishMaker(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(gibberishMaker(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(gibberishMaker(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
+
+
+//   ReferenceError: gibberishMaker is not defined
+
 
 
 // b) Create the function that makes the test pass.
 
+    //----Psuedo code----//
+        //declare a function called gibberishMaker
+        // set a param string
+        // declare a variable called newArray 
+        // used .split to convert the string into an array
+        //declare a variable called newNewArray
+        //use .map to iterate through each value in the array
+        // use conditional statements to determine the values that need replacement
+            // if value === "a", return value = 4
+            // else if value === "e", return value = 3
+            // else if value === "i", return value = 1
+            // else if value === "o", return value = 0
+            // else return value 
+        // use .join to convert the mapped array into a string
+
+
+
+    const gibberishMaker = (string) => {
+    //    let stringTurnedArray = string.split('')
+               let newArray = string.split('')
+               console.log((newArray))
+               let newNewArray= newArray.map ((value) => {
+            if (value.toUpperCase() === "A"){
+                return value = 4 
+            }else if (value.toUpperCase() === "E"){
+                return value = 3
+            }else if (value.toUpperCase() === "I"){
+                return value = 1
+            }else if (value.toUpperCase() === "O"){
+                return value = 0
+            }else{
+                return value 
+            }
+           })
+           return newNewArray.join('')
+        }
+
+ // Test Suites: 1 passed, 1 total
+//  This problem basically took me the entire length of The Curious Case of Benjamin Button to figure out (i took breaks), I'm going to refactor later
+
```

Comment:
Wow - that's a long movie.  Great work!  As always thank you for your function names and comments!  I love reviewing your assessments.